onboarding_documentation/technical_documentation/metamist_docker/metamist_docker_tutorial.md

```diff
@@ -0,0 +1,737 @@
+[FastQE](https://github.com/fastqe/fastqe/) is a fun tool that mimics the output of FastQC (a more 'official' bioinformatic tool) but instead represents the quality scores with emojis. Clearly, FastQE is not intended for use in a production environment. However, it serves as an illustrative example of a tool currently missing from our library. To incorporate it, we need to construct a corresponding Docker image.
+
+Have a go at writing a Dockerfile for FastQE. Have a look at the [images](https://github.com/populationgenomics/images) repo to see how other images are built. **Hint:** You can use the `python:3.10-slim` image as a base image and will install FastQE version 0.3.1 using `pip`.
+
+<details>
+  <summary>Click to see answer</summary>
+
+```Docker
+FROM python:3.10-slim
+
+ARG VERSION=${VERSION:-0.3.1}
+RUN pip install fastqe==${VERSION}
+```
+
+</details>
+<br>
+
+Save the above in a file named `Dockerfile` and push it to the `images` repo, create a PR and as is standard practice in CPG you will have to nominate someone to review this PR so that it can be merged into main and the image can be used by `analysis-runner` when we run the job.
+
+CLI command to build the image:
+
+```bash
+docker build -t fastqe_image:1.0.0 . --platform=linux/amd64
+```
+
+Note that we need to specify the platform as `linux/amd64` because we are building the image on a Mac. If you are building the image on a Linux machine, you can omit the `--platform` flag.
+
+
+
+## 3. Write a script to run FastQE
+
+The basic outline of our script is going to be:
+
+1. Query Metamist for the `SequencingGroup` ID's of the samples we want to run FastQE on
+2. For each `SequencingGroup` ID:
+    - Query Metamist for the location of the fastq files
+    - Build a dictionary mapping the `SequencingGroup` ID to the location of the fastq files
+3. Build a Hail Batch command that uses the FastQE image
+    - Run FastQE on the fastq files
+
+### Task: Write the query to get the `SequencingGroup` ID's of the samples we want to run FastQE on and the location of their reads files
+
+- Tip: first import `gql` and `query` from `metamist.graphql`. Then write the query as a global variable and wrap it in a `gql()` call. Use the GraphQL interface to practice the query before writing it in Python.
+
+<br>
+
+<details>
+  <summary>Click to see answer</summary>
+
+```python
+from metamist.graphql import gql, query
+
+SG_ASSAY_QUERY = gql(
+    """
+    query ($project: String!) {
+        sequencingGroups(project: {eq: $project}) {
+            id
+            assays {
+                id
+                meta
+            }
+        }
+    }
+    """
+)
+```
+
+</details>
+<br>
+
+
+The next step is to iterate through the response of the above query and create a mapping of `SequencingGroup` ID's to the locations of the fastq files.
+
+### Task: Write the function called `get_assays()` to map the `SequencingGroup` ID's to the locations of the fastq files
```

Comment:
A ton of the content here could be separated out into an actual script, which you can reference in part (like with links to specific lines). I think you've probably go this script somewhere for the FQE process. Duplicating that code here makes this a massive document for not much extra benefit.

Personally I'd link out to a fully featured script which is mostly self-documenting, and have a section here just explaining the intent